types/database/channel.d.ts

```diff
@@ -0,0 +1,81 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import {Query, Relation} from '@nozbe/watermelondb';
+import Model, {Associations} from '@nozbe/watermelondb/Model';
+
+import ChannelInfo from '@typings/database/channel_info';
+import ChannelMembership from '@typings/database/channel_membership';
+import Draft from '@typings/database/draft';
+import GroupsInChannel from '@typings/database/groups_in_channel';
+import MyChannel from '@typings/database/my_channel';
+import MyChannelSettings from '@typings/database/my_channel_settings';
+import Post from '@typings/database/post';
+import PostsInChannel from '@typings/database/posts_in_channel';
+import Team from '@typings/database/team';
+import User from '@typings/database/user';
+
+/**
+ * The Channel model represents a channel in the Mattermost app.
+ */
+export default class Channel extends Model {
+    /** table (entity name) : Channel */
+    static table: string;
+
+    /** associations : Describes every relationship to this entity. */
+    static associations: Associations;
+
+    /** create_at : The creation date for this channel */
+    createAt: number;
+
+    /** creator_id : The user who created this channel */
+    creatorId: string;
+
+    /** delete_at : The deletion/archived date of this channel */
+    deleteAt: number;
+
+    /** display_name : The channel display name (e.g. Town Square ) */
+    displayName: string;
+
+    /** is_group_constrained : If group is restricted to certain users/teams only */
```

Comment:
```suggestion
    /** is_group_constrained : If group is restricted to certain members of a group only */
```

@hahmadia can  you confirm my suggestion is valid?

Reply:
If a channel is group constrained, it means that only the members in those groups have access to that channel.
```suggestion
    /** is_group_constrained : If a channel is restricted to certain groups only (hence only members of that group are allowed to access this channel) */
```

Reply:
You may edit as you deem necessary. I was trying to just give a bit more explanation on it.